R/deploymentTarget.R

```diff
@@ -1,106 +1,277 @@
-# calculate the deployment target based on the passed parameters and
-# any saved deployments that we have
-deploymentTarget <- function(recordPath = ".",
-                             appId = NULL,
-                             appName = NULL,
-                             appTitle = NULL,
-                             envVars = NULL,
-                             account = NULL,
-                             server = NULL,
-                             forceUpdate = FALSE,
-                             error_call = caller_env()) {
+# Discover the deployment target given the passed information.
+#
+# Returns a list containing a deployment record and the account details to use
+# when performing the deployment.
+#
+# When appId is provided, it must identify an existing application. The
```

Comment:
I'd maybe spread these amongst the existing implementation?

Reply:
I believe this is already the case -- each of the `findDeploymentTargetBy*` helpers explains its responsibility. This is the roll-up.